x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -307,6 +334,12 @@ public ActionRequestValidationException validate() {
             if (threadsPerAllocation < 1) {
                 validationException.addValidationError("[" + THREADS_PER_ALLOCATION + "] must be a positive integer");
             }
+            ActionRequestValidationException autoscaleException = adaptiveAllocationsSettings == null
+                ? null
+                : adaptiveAllocationsSettings.validate();
```

Comment:
For simplicity, the behaviour is:
- the initial number of allocation is `number_of_allocations` (the existing setting)
- the scaler will adjust it to something in the range `[min, max]`

I don't think it's worth the effort to enforce `num` to be in the range of `[min, max]` in case `enabled=true`.

Reply:
I don't the exact syntax and spelling but a start deployment request looks something like:

```
{
    number_of_allocations: 1,
    threads_per_allocation: 1,
    autoscaling: {
        enabled: true,
        min : 1
        max: 8
   }
}
```

If the user was to make a request where `number_of_allocations` is outside the [min, max] range then they have made an error and should be told that the values are contradictory. This is helpful to the user to prevent mistakes.

```
{
    number_of_allocations: 12,
    threads_per_allocation: 1,
    autoscaling: {
        enabled: true,
        min : 1
        max: 8
   }
}
```

Reply:
OK, guess you're right. I've updated the API like this: when adaptive allocations is enabled, setting `number_of_allocations` is not allowed. It's useless anyway, because the system will undo your changes in 10 secs.